src/core/utils/time/timeUtils.ts

```diff
@@ -0,0 +1,96 @@
+import {
+  DAY_IN_HRS,
+  DAY_IN_S,
+  HOUR_IN_S,
+  MINUTE_IN_S,
+  SECOND_IN_MS
+} from "./timeConstants";
+import {CountdownResults} from "./timeTypes";
+import {CountdownItem} from "../../../countdown/Countdown";
+
+function calculateRemainingTime(target: Date): CountdownResults {
+  const delta = target.getTime() - new Date().getTime();
+  const deltaInSeconds = delta / SECOND_IN_MS;
+
+  return {
+    delta,
+    days: Math.floor(deltaInSeconds / DAY_IN_S),
+    hours: Math.floor((deltaInSeconds / HOUR_IN_S) % DAY_IN_HRS),
+    minutes: Math.floor((deltaInSeconds % HOUR_IN_S) / MINUTE_IN_S),
+    seconds: Math.floor((deltaInSeconds % HOUR_IN_S) % MINUTE_IN_S)
+  };
+}
+
+function generateCountdownItems(
+  {alwaysShowSeconds}: {alwaysShowSeconds: boolean},
+  countdownData: CountdownResults
+): CountdownItem[] {
+  const maxDigit = 9;
+  let items: CountdownItem[];
+
+  if (countdownData.days >= 1) {
+    items = [
+      {
+        id: "days",
+        count:
+          countdownData.days <= maxDigit ? `0${countdownData.days}` : countdownData.days
```

Comment:
We can just use `padStart` method here instead of having to check against `maxDigit`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart
```suggestion
          countdownData.days.padStart(2, "0")
```